src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -7858,56 +7875,258 @@ def apply(
-                func,
-                column_index,
-                input_types,
-                return_type,
-                udf_args=args,
-                udf_kwargs=kwargs,
-                session=self._modin_frame.ordered_dataframe.session,
-            )
+                agg_func = wrapped_func
+            else:
+                agg_func = func
+
+            # Accumulate indices of the column results.
+            col_result_indexes = []
+            # Accumulate "is scalar" flags for the column results.
+            col_result_scalars = []
+
+            for (
+                data_column_pandas_label,
+                data_column_snowflake_quoted_identifier,
+            ) in zip(
+                frame.data_column_pandas_labels,
+                frame.data_column_snowflake_quoted_identifiers,
+            ):
+                data_col_frame = InternalFrame.create(
+                    ordered_dataframe=frame.ordered_dataframe,
+                    data_column_snowflake_quoted_identifiers=[
+                        data_column_snowflake_quoted_identifier
+                    ],
+                    data_column_pandas_labels=[data_column_pandas_label],
+                    data_column_pandas_index_names=frame.data_column_pandas_index_names,
+                    data_column_types=None,
+                    index_column_snowflake_quoted_identifiers=frame.index_column_snowflake_quoted_identifiers,
+                    index_column_pandas_labels=frame.index_column_pandas_labels,
+                    index_column_types=None,
+                )
+
+                data_col_qc = SnowflakeQueryCompiler(data_col_frame).groupby_apply(
+                    by=[],
+                    agg_func=agg_func,
+                    axis=0,
+                    groupby_kwargs={"as_index": False, "dropna": False},
+                    agg_args=args,
+                    agg_kwargs=kwargs,
+                    series_groupby=True,
+                    force_single_group=True,
+                    force_list_like_to_series=True,
+                )
+
+                data_col_result_frame = data_col_qc._modin_frame
+
+                # Set the index name on the result
+                data_col_result_frame = InternalFrame.create(
+                    ordered_dataframe=data_col_result_frame.ordered_dataframe,
+                    data_column_snowflake_quoted_identifiers=data_col_result_frame.data_column_snowflake_quoted_identifiers,
+                    data_column_pandas_labels=[data_column_pandas_label],
+                    data_column_pandas_index_names=data_col_frame.data_column_pandas_index_names,
+                    data_column_types=None,
+                    index_column_snowflake_quoted_identifiers=data_col_result_frame.index_column_snowflake_quoted_identifiers,
+                    index_column_pandas_labels=data_col_result_frame.index_column_pandas_labels,
+                    index_column_types=None,
+                )
```

Comment:
Why not use `set_index_names` instead?

Reply:
It also needs to set the data_column_pandas_label otherwise it won't have the expected result.  I updated the comment to mention this.